core/src/tangram.cpp

```diff
@@ -548,4 +559,9 @@ void setupGL() {
     while (Error::hadGlError("Tangram::setupGL()")) {}
 }
 
+void runOnMainLoop(std::function<void()> _task) {
+    std::lock_guard<std::mutex> lock(m_tasksMutex);
+    m_tasks.emplace(std::move(_task));
```

Comment:
`m_tasks` is emptied on every main loop. Currently the only tasks that go into this queue are lambdas that delete `TileBuilder` instances. Enqueueing those instances for deletion on the main thread is the fastest way that we can safely delete them.